backend/api/v1/setting_service.go

```diff
@@ -162,6 +165,18 @@ func (s *SettingService) UpdateSetting(ctx context.Context, request *v1pb.Update
 	if existedSetting == nil && !request.AllowMissing {
 		return nil, status.Errorf(codes.NotFound, "setting %s not found", settingName)
 	}
+	// audit log.
+	if setServiceData, ok := common.GetSetServiceDataFromContext(ctx); ok {
+		v1pbSetting, err := s.convertToSettingMessage(ctx, existedSetting)
+		if err != nil {
+			slog.Warn("audit: failed to convert to v1.Setting", log.BBError(err))
+		}
+		p, err := anypb.New(v1pbSetting)
+		if err != nil {
+			slog.Warn("audit: failed to convert to anypb.Any", log.BBError(err))
+		}
+		setServiceData(p)
```

Comment:
I prefer to log it after setting update (after line 600), then we also don't need to convert it to v1 api